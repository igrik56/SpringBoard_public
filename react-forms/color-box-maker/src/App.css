.App {
  text-align: center;
}

div{
  display: block;
  margin-top: 1rem;
}