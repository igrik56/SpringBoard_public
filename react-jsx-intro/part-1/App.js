const FirstComponent = () =>{
    return <h1>My very first component</h1>
}

const NamedComponent = (props) =>{
    return (
        <p>My name is {props.name}</p>
    )
}

const App = () => {
    return(
        <div>
            <FirstComponent />
            <NamedComponent name="Mike"/>
        </div>
    )
}

ReactDOM.render(<App/>, document.getElementById("root"))