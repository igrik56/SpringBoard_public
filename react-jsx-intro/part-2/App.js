const Tweet = (props) => {
    return (
        <div>
            <h3>{props.name}</h3>
            <h4>@{props.username}</h4>
            <span>{props.date}</span>
            <p>{props.message}</p>
        </div>
    )
}

const App = () => {
    return(
        <div>
            <Tweet name="Mike" username="mikeWazowski" date="01/01/2023" message="Happy New Year!"/>
        </div>
    )
}

ReactDOM.render(<App/>, document.getElementById("root"))