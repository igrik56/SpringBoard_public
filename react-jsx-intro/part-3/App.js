const App = () => {
    return(
        <div>
            <Person name="mikeWazowski" age="28" />
            <Person name="Sully" age={29} hobbies={["bowling", "watching tv", "drinking beer"]} />
        </div>
    )
}

ReactDOM.render(<App />, document.getElementById("root"))