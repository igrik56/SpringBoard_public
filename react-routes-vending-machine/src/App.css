.App {
  text-align: center;
  align-items: center;
}

.App img{
  margin: 5rem;
  scale: 1.2;
}