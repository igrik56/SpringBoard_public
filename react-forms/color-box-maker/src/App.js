import React from 'react';
import BoxList from './BoxList';
import './App.css';

function App() {
  return (
    <div className="App">
     <BoxList />
    </div>
  );
}

export default App;
