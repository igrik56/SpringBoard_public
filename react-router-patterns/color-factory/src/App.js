import RoutesColorFactory from './RoutesColorFactory'
import './App.css';

function App() {
  return (
    <div className="App">
      <RoutesColorFactory />
    </div>
  );
}

export default App;
