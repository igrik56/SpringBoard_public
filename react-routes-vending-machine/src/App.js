import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Navbar from './Navbar'
import Home from './Home'
import Cola from './Cola'
import Chips from './Chips'
import Pudding from './Pudding'
import Kolobochek from './Kolobochek'

import './App.css';


function App() {
  return (
    <div className="App">
     <BrowserRouter>
        <Navbar />
        <Routes>
          <Route exact path = '/' element={<Home />} />
          <Route exact path = '/Cola' element={<Cola/>} />
          <Route exact path = '/Chips' element ={<Chips />} />
          <Route exact path = '/Pudding' element={<Pudding />} />
          <Route exact path = '/Kolobochek' element={<Kolobochek />} />
          </Routes>      
     </BrowserRouter>
    </div>
  );
}

export default App;
