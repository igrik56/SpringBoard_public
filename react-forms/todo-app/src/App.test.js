import { render } from '@testing-library/react';
import App from './App';
import React from 'react';

test('renders learn react link', () => {
  render(<App />);
});

it('snapshot test', function() {
  const { asFragment } = render(<App />)
  expect(asFragment()).toMatchSnapshot()
})
