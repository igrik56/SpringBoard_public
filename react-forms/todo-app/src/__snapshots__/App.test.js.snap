// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div>
        <form>
          <div>
            <label
              for="task"
            >
              Task: 
            </label>
            <input
              id="task"
              name="task"
              type="text"
              value=""
            />
            <button
              id="addNewTodo"
            >
              Add a task to the list
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
